test/CMakeLists.txt

```diff
@@ -48,22 +50,54 @@ set(XEUS_CPP_TESTS
 
 add_executable(test_xeus_cpp ${XEUS_CPP_TESTS})
 
-if (APPLE)
-    set_target_properties(test_xeus_cpp PROPERTIES
-        MACOSX_RPATH ON
+if(EMSCRIPTEN)
+    target_link_libraries(test_xeus_cpp PRIVATE xeus-cpp-static doctest::doctest)
+
+    target_compile_options(test_xeus_cpp
+        PUBLIC "SHELL: -fexceptions"
+    )
+
+    target_link_options(test_xeus_cpp
+        PUBLIC "SHELL: -fexceptions"
+        PUBLIC "SHELL: -s MAIN_MODULE=1"
+        PUBLIC "SHELL: -s ASSERTIONS=0"
+        PUBLIC "SHELL: -s ALLOW_MEMORY_GROWTH=1"
+        PUBLIC "SHELL: -s EXIT_RUNTIME=1"
```

Comment:
Needed for doctests to exit smoothly. Otherwise hangs after finishing running the tests.